code/modules/antagonists/heretic/heretic_antag.dm

```diff
@@ -142,13 +145,22 @@
 			if(!ispath(researched_path))
 				CRASH("Heretic attempted to learn non-heretic_knowledge path! (Got: [researched_path])")
 
-			if(initial(researched_path.cost) > knowledge_points)
-				return TRUE
+			// If side path and has path points, buy!
+			var/coupon = FALSE
+			if((initial(researched_path.route) == PATH_SIDE )&& (side_path_points > 0))
+				coupon = TRUE
+			// else try normal purchase
+			else if(initial(researched_path.cost) > knowledge_points)
+				return
+
 			if(!gain_knowledge(researched_path))
 				return TRUE
 
 			log_heretic_knowledge("[key_name(owner)] gained knowledge: [initial(researched_path.name)]")
```

Comment:
Can you update this to indicate whether a coupon was used
Just something like `"gained knowledge (via free sidepath point):"`